service/hyperOutportBlock/blockService.go

```diff
@@ -49,3 +56,33 @@ func (bs *Service) GetHyperOutportBlockByNonce(ctx context.Context, req *data.Bl
 
 	return hyperOutportBlock, nil
 }
+
+func (bs *Service) HyperOutportBlockStream(_ *empty.Empty, stream data.HyperOutportBlockService_HyperOutportBlockStreamServer) error {
+	for {
+		select {
+		case <-bs.ctx.Done():
+			if len(*bs.blocksChannel) == 0 {
+				return nil
+			}
+
+			for len(*bs.blocksChannel) > 0 {
+				block := <-*bs.blocksChannel
+
+				err := stream.Send(block)
+				if err != nil {
+					return fmt.Errorf("failed to send hyperOutportBlock: %w", err)
+				}
+			}
+		// Exit on stream context done.
+		case <-stream.Context().Done():
+			return nil
+		default:
+			block := <-*bs.blocksChannel
+
+			err := stream.Send(block)
+			if err != nil {
```

Comment:
can you test what happens if the grpc client (stream consumer) closes/looses connection temporarily and then reconnects will it open a new stream, and if so will can it continue with streaming the missed blocks?

maybe we should have multiple retrials as well
seems that if any error occurs, the process loop for streaming the blocks immediately stops.

Reply:
just took a look athis use case now. It does open a new stream but it seems some blocks are lost. let me see what I can do.